File: 프로그래머스/파이썬/Lv2/소수찾기.py
from itertools import permutations
import math


def solution(numbers):
    answer = 0

    tmp = []

    def check(x):
        for i in range(2, int(math.sqrt(x)) + 1):
            if x % i == 0:
                return False
        return True

    for i in range(1, len(numbers) + 1):
        for i in permutations(numbers, i):
            res = int("".join(list(i)))
            if res >= 2 and res not in tmp:
                if check(res):
                    tmp.append(res)

    return len(tmp)

File: 프로그래머스/JavaScript/소수 찾기.js
function solution(n) {
    var answer = n - 1;
    var arr = new Array(n+1).fill(1);

    for (let i = 2; i <= Math.sqrt(n); i++){
        if (arr[i]){
            for (let j = i ** 2; j <= n; j+=i) {
                if (arr[j]) {
                    answer--;
                    arr[j] = 0;
                }
            }
        }
    }
    return answer;
}